ipaclient/install/client.py

```diff
@@ -2907,6 +2907,14 @@ def _install(options):
 
         # Check that nss is working properly
         if not options.on_master:
+            user = options.principal
+            if user is None:
+                user = "admin"
+                root_logger.info("Principal is not set when enrolling with OTP"
+                                 "; using principal '%s' for 'getent passwd'",
+                                 user)
+            else:
+                user = user.split("@")[0]
```

Comment:
Hi @Tiboris ,
I still have a nitpick. With this call to split, the code is actually loosing some information. For instance, if the options.principal was myuser@bogusdomain.com we end up with user = myuser and run getent passwd myuser@ipadomain.com (ie we replace bogusdomain with ipadomain.com).
I would rather do the opposite, i.e make sure that the "user" variable contains a kerberos id with the format name@domain (if it contains only name, then add @cli_domain) or admin@cli_domain, and call "getent passwd user"  instead of "getent passwd user@cli_domain".